stdlib/sys.mli

```diff
@@ -143,9 +143,11 @@ val max_string_length : int
 (** Maximum length of strings and byte sequences. *)
 
 val max_array_length : int
-(** Maximum length of a normal array.  The maximum length of a float
-    array is [max_array_length/2] on 32-bit machines and
-    [max_array_length] on 64-bit machines. *)
+(** Maximum length of a normal array. *)
+
+val max_float_array_length : int
+(** Maximum length of a float array: [max_array_length/2] on 32-bit
```

Comment:
This is true only when flat float arrays are not disabled (and hopefully in the future they will commonly be).  The documentation would be relevant for another value `max_floatarray_length`, which could also be useful (although I believe we would be better of with String.max_length, Bytes.max_length, Float.Array.max_length, etc).